polkadot/node/collation-generation/src/lib.rs

```diff
@@ -596,25 +537,47 @@ async fn construct_and_distribute_receipt(
 		hrmp_watermark: collation.hrmp_watermark,
 	};
 
-	let ccr = CandidateReceipt {
-		commitments_hash: commitments.hash(),
-		descriptor: CandidateDescriptor {
-			signature: key.sign(&signature_payload),
-			para_id,
-			relay_parent,
-			collator: key.public(),
-			persisted_validation_data_hash,
-			pov_hash,
-			erasure_root,
-			para_head: commitments.head_data.hash(),
-			validation_code_hash,
+	let receipt = if v2_receipts {
+		let ccr = CommittedCandidateReceiptV2 {
+			descriptor: CandidateDescriptorV2::new(
+				para_id,
+				relay_parent,
+				core_index,
```

Comment:
We can simplify the collator code and also avoid checking the core index here if we use `commitments.committed_core_index()`. We need to however use the PreparedCollation core index if there is no core index commitment.

Reply:
how would this simplify the collator code?

Reply:
I mean we don't have to look into the commitments because we just do it here when assembling the committed candidate receipt.